### ISSUE TITLE
Base classes for "recalibration" techniques

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/rail_base#27]
There's been a lot of discussion of "recalibration" methods for n(z) and p(z)s, and they would best fit into the current framework as a summarizer accepting the n(z) output of another summarizer plus other information as input and outputting n(z) and an estimator accepting the p(z) output of another estimator plus other information as input and outputting p(z)s, respectively. Our Estimator and Summarizer stage superclasses are defined by what kinds of inputs they need, so this issue is for making those base classes and at least one "dummy" subclass for each so there can be placeholder stages for these procedures in pipelines currently under development.

**Comment by sschmidt23** [comment]
I think we should have the discussion on the expected form that this will take before we think about the class structure.  This could be something that better lives outside of RAIL, or it might all fit within RAIL, but I don't think that we'll know for sure until we have a more concrete pipeline diagram written down (which should be done with outside groups using tomographic N(z) involved in the discussion as well).

**Comment by aimalz** [comment]
Fair point, I added the "question" label so we know to discuss it broadly before taking any action on implementation.

**Comment by eacharles** [comment]
The interfaces needed here are covered generically by the PzInformer (allowing you to make p(z) using p(z) as and input, and SZPZSummarizer (allowing you to make n(z)) and the sompz classes calibration). 

We discussed this in the rail-tag up and agreed to close this issue.
